### PR TITLE
[#372] Implement XtendHoverInEditorTest javadoc hovering test cases.

### PR DESCRIPTION
**Pull request by miklossy**
Signed-off-by: Tamas Miklossy <miklossy@itemis.de>

### CODE REVIEW

File: org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/hover/XtendHoverInEditorTest.xtend [hunk 1]
```diff
@@ -113,4 +116,103 @@ class XtendHoverInEditorTest extends AbstractXtendUITestCase {
...
+		'''.hasHoverOverJavaType("Object", '''Class <code>Object</code> is the root of the class hierarchy.''')
+	}
+
+	@Test def testHoverOfJavaTypeInJavadocSee() {
+		'''
+			/**
+			 * @see Object
+			 */
+			class Foo {}
+		'''.hasHoverOverJavaType("Object", '''Class <code>Object</code> is the root of the class hierarchy.''')
```
Comment (tivervac) [inline org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/hover/XtendHoverInEditorTest.xtend]: Might be good to abstract the Object assert into a separate method e.g. "test".assertHasObjectHover

File: org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/hover/XtendHoverInEditorTest.xtend [hunk 2]
```diff
@@ -113,4 +116,103 @@ class XtendHoverInEditorTest extends AbstractXtendUITestCase {
...
+		"Bar.xtend".createFile('''
+			/**
+			 * Documentation of the Bar class.
+			 */
+			class Bar {}
+		''')
+		
+		'''
+			/**
+			 * @see Bar
```
Comment (tivervac) [inline org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/hover/XtendHoverInEditorTest.xtend]: A test with a fully qualifiedname might be useful as well e.g. `@see org.eclipse.xtend.ide.tests.hover.Bar`

File: org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/hover/XtendHoverInEditorTest.xtend [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (tivervac) [inline org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/hover/XtendHoverInEditorTest.xtend]: Does this new feature also support method references? e.g. `@see Bar#foo` This should refer to the `foo` method in the `Bar` class

**Comment by miklossy** [comment]
Thanks for the feedback @tivervac ! I updated the PR according to your remarks.

The method references in javadocs are not yet supported, but I would say that that is out of scope of this PR.

File: org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/hover/XtendHoverInEditorTest.xtend (hunk 3 above)
Comment (tivervac) [reply org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/hover/XtendHoverInEditorTest.xtend]: Not yet supported.

**Comment by tivervac** [comment]
LGTM

**Comment by cdietrich** [comment]
i assume the tests are failing cause we dont have upstream detection

**Comment by miklossy** [comment]
@cdietrich is right, since the tests are successful on my computer.

**Comment by cdietrich** [comment]
thx @miklossy

**Pull request merged**